File: 509.fibonacci-number.kt
/*
 * @lc app=leetcode id=509 lang=kotlin
 *
 * [509] Fibonacci Number
 */

// @lc code=start
class Solution {
    fun fib(n: Int): Int {
        if (n <= 1) return n
        var a = 0
        var b = 1
        for (i in 2..n) {
            val temp = a + b
            a = b
            b = temp
        }
        return b
    }
}
// @lc code=end

File: 509.fibonacci-number.py
#
# @lc app=leetcode id=509 lang=python3
#
# [509] Fibonacci Number
#

# @lc code=start
class Solution:
    def fib(self, n: int) -> int:
        if n < 2:
            return n
        a, b = 0, 1
        for _ in range(1, n):
            a, b = b, a + b
        return b
# @lc code=end
